charmhub/refresh.go

```diff
@@ -479,11 +470,3 @@ func ExtractConfigInstanceKey(cfg RefreshConfig) string {
 	}
 	return ""
 }
-
-var logger = loggo.GetLoggerWithLabels("juju.charmhub", corelogger.CHARMHUB)
-
-func logAndReturnError(err error) error {
-	err = errors.Trace(err)
-	logger.Errorf(err.Error())
-	return err
-}
```

Comment:
Too many times, errors get lost going across some wire. We explicitly added this code due to problems debugging. Please add back.

Reply:
Per discussion, I've added this back. I was going to make it use the logger provided to `Config.Logger`, but most of the calls to `logAndReturnError` are from top-level `RefreshConfig`-creating functions like `RefreshOne` that don't have that context. So I just added an explanatory comment.